src/Tree/Tree.cpp

```diff
@@ -1457,6 +1626,235 @@ FLOAT Tree<ndim,ParticleType,TreeCell>::ComputeWorkInBox
+template <int ndim, template<int> class ParticleType, template<int> class TreeCell>
+int Tree<ndim,ParticleType,TreeCell>::GetSizeOfExportedParticleData(int Nparticles) const {
+  typedef typename ParticleType<ndim>::HandlerType::DataType StreamlinedPart;
+  return Nparticles*sizeof(StreamlinedPart) ;
+}
+//=================================================================================================
+//  Tree::GetSizeOfExportedCellData
+/// Get size needed for packed cell data
+//=================================================================================================
+template <int ndim, template<int> class ParticleType, template<int> class TreeCell>
+int Tree<ndim,ParticleType,TreeCell>::GetSizeOfExportedCellData(int Ncell) const {
+  typedef typename TreeCell<ndim>::HandlerType::DataType StreamlinedCell;
+  return Ncell * sizeof(StreamlinedCell) ;
+}
+//=================================================================================================
+//  Tree::GetSizeOfReturnedParticleData
+/// Get size needed for packed particle data returned after force calculation
+//=================================================================================================
+template <int ndim, template<int> class ParticleType, template<int> class TreeCell>
+int Tree<ndim,ParticleType,TreeCell>::GetSizeOfReturnedParticleData(int Nparticles) const {
+  typedef typename ParticleType<ndim>::HandlerType::ReturnDataType StreamlinedPart;
+  return sizeof(StreamlinedPart) * Nparticles ;
+}
+//=================================================================================================
+//  Tree::GetSizeOCellData
+/// Get size needed for packed cell data returned after force calculation
+//=================================================================================================
+template <int ndim, template<int> class ParticleType, template<int> class TreeCell>
+int Tree<ndim,ParticleType,TreeCell>::GetSizeOfReturnedCellData(int Ncell) const {
+  return sizeof(double) * Ncell ;
+}
+
+//=================================================================================================
+//  Tree::AddWorkCost
+/// Add the work done for active particles to the tree cells
+//=================================================================================================
+template <int ndim, template<int> class ParticleType, template<int> class TreeCell>
+void Tree<ndim,ParticleType,TreeCell>::AddWorkCost(vector<TreeCellBase<ndim> >& celllist,
+                                                   double twork, int& Nactivetot_out) {
+   int cactive = celllist.size();
+   int Nactivetot=0;
+   for (int cc=0; cc<cactive; cc++) Nactivetot += celllist[cc].Nactive;
+   for (int cc=0; cc<cactive; cc++) {
+     int c = celllist[cc].id;
+     celldata[c].worktot += twork*(DOUBLE) celldata[c].Nactive / (DOUBLE) Nactivetot;
+   }
+
+   Nactivetot_out =  Nactivetot ;
+}
```

Comment:
Could do, but this avoided some warnings when the value was set but not used in the density/force routines. It's only needed when OUTPUT_ALL is defined after all.